spec/scry/completion_provider_spec.cr

```diff
@@ -128,9 +128,20 @@ module Scry
 
         it_completes "Node.", %w(new)
 
-        it_completes "a = A.new
+        it_completes "a = B.new
                       a.", %w(method)
       end
     end
+
+    context "classname completion" do
+      _kind_ = CompletionItemKind::Class
+      it_completes("A", %w(Array Atomic ArgumentError AtExitHandlers))
```

Comment:
This means that this spec rely on external code... A change to the stdlib, an addition/removal of a class starting with `A` will break this spec...

There are other problems with how the specs currently works, like:
- You can't add a description of what that spec is actually spec-ing
- They rely on external code (see above)
- They are not self-contained: by just reading the spec you don't directly have all the data to understand it (you have to always lookup other files, AND they are shared between many specs)

This should probably go to a new issue though. _I'm just too tired right now_ ^^

Reply:
1. The test will always break with changes to stdlib, we are testing the autocomplete on it I don't think it's something we should worry about.
2. What other files do you have to check to understand the specs? My idea for this file was to have tests that almost did an end-to-end test on the completion features and not dependant on the actual implementation.